### PR TITLE
Updating Release Notes for 1.300043.0

### PR DESCRIPTION
**Pull request by mitali-salvi**
# Description of the issue
Updating the release notes.

# Description of changes
_How does this change address the problem?_

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
_Describe what tests you have done._

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

**Review by zhihonl — DISMISSED** [review]

File: RELEASE_NOTES [hunk 1]
```diff
@@ -1,3 +1,15 @@
+========================================================================
+Amazon CloudWatch Agent 1.300043.0 (2024-08-06)
+========================================================================
+Bug Fixes:
+* [Metrics/Plugin] Drop original metrics support added for statsd, collectd and ethtool metrics
+* [Logs/Windows Event] Add windows event log service restart detection and resubscribe
+
+Enhancements:
+* Add cumulative to delta conversion for JMX metrics. by @jefchien
```
Comment (jefchien) [inline RELEASE_NOTES:9]:
```suggestion
* [Metrics/JMX] Add cumulative to delta conversion for JMX metrics
```

File: RELEASE_NOTES [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+Amazon CloudWatch Agent 1.300043.0 (2024-08-06)
+========================================================================
+Bug Fixes:
+* [Metrics/Plugin] Drop original metrics support added for statsd, collectd and ethtool metrics
+* [Logs/Windows Event] Add windows event log service restart detection and resubscribe
+
+Enhancements:
+* [Metrics/JMX] Add cumulative to delta conversion for JMX metrics
+* [ApplicationSignals] Rename and group application signals attributes
+* Upgrade OTEL Contrib to v0.103.0
```
Comment (jefchien) [inline RELEASE_NOTES:11]:
```suggestion
* Upgrade OTEL dependencies to v0.103.0
```

Comment (mitali-salvi) [reply RELEASE_NOTES:11]: I followed the same pattern as here -  https://github.com/aws/amazon-cloudwatch-agent/blob/main/RELEASE_NOTES#L57

Comment (jefchien) [reply RELEASE_NOTES:11]: Yeah I'm a bit surprised that we worded it that way before.

**Review by jefchien — APPROVED** [review]

**Review by zhihonl — APPROVED** [review]

**Pull request merged**